ERCS/erc-7272.md

```diff
@@ -0,0 +1,142 @@
+
+1. A user interacts with the DeFi's off-chain service, providing sufficient input for the off-chain service to ensure the user meets its criteria (for example, authenticates the user and/or make sure they possess valid credentials)
+2. If authorization is granted, an EAT is issued to the user
+3. The user then interacts with the gated smart contract function within the specified period of time passing the EAT as part of the transaction
+4. The EAT is verified on-chain
+
+![Transaction authorization flow using an EAT](../assets/erc-7272/EAT_transaction_auth_flow.png)
+
+An Ethereum Access Token MUST guarantee granular access control by binding it to specific parameters upon issuance. Then, on-chain EAT verification ensures that:
+
+- The function being called is the expected one
+- The function parameters are the expected ones
+- The function caller is the expected one
+- The function is being called in the authorized timeframe (i.e checking that the EAT is not expired)
+- The smart contract being called is the expected one
+- The authorization has been given by a valid issuer, i.e the EAT has been signed by one of the expected issuers
+
+### Structure of an Ethereum Access Token
+
+An Ethereum Access Token is composed of a signature and expiry.
+
+```
+{
+ uint8 v,
+ bytes32 r,
+ bytes32 s,
+ uint256 expiry
+}
+```
+
+The signature is obtained using the typed structured data hashing and signing standard (EIP-712), signing over the following EAT payload:
+
+```
+struct AccessToken {
+    uint256 expiry;
+    FunctionCall functionCall;
+}
+
+struct FunctionCall {
+    bytes4 functionSignature;
+    address target;
+    address caller;
+    bytes parameters;
+}
+```
+
+- **expiry**: unix timestamp, expected to be before `block.timestamp`
+
+`FunctionCall` parameters correspond to the following:
+
+- **functionSignature**: identifier for the function being called, expected to match `msg.sig`
+- **target**: address of the target contract being called
+- **caller**: address of the current caller - expected to match `msg.sender`
+- **parameters**: `calldata` after stripping off the first parameters, namely `v`,`r`, `s` and `expiry`
+
+### EAT Verification
+
+On chain, the reference implementation uses two contracts: an `AccessTokenConsumer` which is inherited by contracts needing to permission some of its functions and an `AccessTokenVerifier` which is responsible for verifying EATs.
+
+The `AccessTokenConsumer` contract calls the `AccessTokenVerifier` to verify the integrity of an EAT.
+
+The `verify()` function of the `AccessTokenVerifier` takes a signature and an `AccessToken` as input, verifies that the token is not expired, attempts to recover the signer from the signature and the reconstructed EIP-712 digest, and verifies that the signer is a valid, expected signer.
+
+Please see the [reference implementation](../assets/erc-7272/AccessTokenVerifier.sol) for an example of how this can be performed.
```

Comment:
```suggestion
Please see the [reference implementation](../assets/eip-7272/AccessTokenVerifier.sol) for an example of how this can be performed.
```